framework/src/modules/chain/rebuilder.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const { loadBlocksWithOffset } = require('./blocks');
+
+class Rebuilder {
```

Comment:
Maybe we could explain what this class is responsible for here as a JSDoc comment?